src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java

```diff
@@ -105,6 +106,29 @@ public Dataset find(Object pk) {
         return em.find(Dataset.class, pk);
     }
 
+    public Dataset findDeep(Object pk) {
```

Comment:
Can you add a comment about when to use this and what it does? If I understand - all of these left-join-fetch hints (and the left-join-fetch in the query) are optimizations to get everything on one call rather than relying on lazy fetching later?